lib/components/term.js

```diff
@@ -234,12 +292,20 @@ export default class Term extends React.PureComponent {
     return !e.catched;
   }
 
-  componentWillReceiveProps(nextProps) {
+  async componentWillReceiveProps(nextProps) {
     if (!this.props.cleared && nextProps.cleared) {
       this.clear();
     }
+
     const nextTermOptions = getTermOptions(nextProps);
 
+    if (nextProps.bellSoundURL) {
+      const value = await getFileData(nextProps.bellSoundURL);
+      if (value) {
+        nextTermOptions.bellSound = value;
```

Comment:
`bellSound` option seems unused 🤔

Reply:
I imagine it isn't the most important feature out of everything, but the default sound provided by Xterm only makes the feature less desirable 😆. The aim was a mix between personal desire (I like having a bell sound) and completeness. I wanted to remove one more thing for someone to complain about 😀

Reply:
To be clear: I LOVE your PR. Seriously, it will be useful and magnify what Hyper is: configurable and Hackable.

What I was trying to say is: You assign `value` to `nextTermOptions.bellSound` but it doesn't seem to be used anywhere. I think that you forgot to add it to Terminal options object like you did for `bellStyle`.

Reply:
Thank you! It's not often I find time to contribute to open source and figured this was an interesting and bite-sized task to handle (bonus points for taking a task that hasn't been reported yet 😆). So thank you for taking the time to evaluate this PR; it's quite motivating as I look forward to contributing where I can.

Okay, so it is a little ambiguous on when it gets used. In the lines just below, `Object.keys` ends up evaluating each option in `nextTermOptions` and filters out only those that have changed which will then call the `setOption` method on the terminal for each option that has changed. 

I originally was going to place the logic in the same area as the `bellStyle` change, but it would end up modifying the nature of `getTermOptions` into an asynchronous function which would create a bit of noise in all the areas `getTermOptions` is being called. I was looking for the least disruptive change, but it's more important that it is done right for the project so ill dig deeper to find where it would fit best. 

Considering @albinekb comments, there would be a way of utilizing `fs.readFileSync()` and drop the "remote url" feature due to valid concerns raised in that idea. On the other hand, `fs.readFileSync()` is usually frowned upon and I don't generally gravitate towards. I do want to take a look at optimization too so we would only have to read the file once, even from local file system, and have this be set for all terminal sessions.

Reply:
Sounds great 👍